### PR TITLE
INTMDB-299: Added snapshot export bucket support

### PR DESCRIPTION
**Pull request by abner-dou**
## Description

Added snapshot export buckets support

Link to any related issue(s): [INTMDB-299](https://jira.mongodb.org/browse/INTMDB-299)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] This change requires a documentation update

## Required Checklist:

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: mongodbatlas/cloud_provider_snapshot_export_buckets.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+	TotalCount int                                  `json:"totalCount,omitempty"` // Count of the total number of items in the result set. It may be greater than the number of objects in the results array if the entire result set is paginated.
+}
+
+// CloudProviderSnapshotExportBucket represents one cloud provider snapshot export buckets.
+type CloudProviderSnapshotExportBucket struct {
+	ID            string `json:"_id,omitempty"`           // Unique identifier of the S3 bucket.
+	ProjectID     string `json:"projectId,omitempty"`     // Unique 24-hexadecimal digit string identifying the project.
+	BucketName    string `json:"bucketName,omitempty"`    // Name of the bucket that the role ID is authorized to access.
+	CloudProvider string `json:"cloudProvider,omitempty"` // Name of the provider of the cloud service where Atlas can access the S3 bucket. Atlas only supports AWS.
+	IamRoleID     string `json:"iamRoleId,omitempty"`     // Unique identifier of the role that Atlas can use to access the bucket. If necessary, use the UI or API to retrieve the role ID. You must also specify the bucketName.
```
Comment (gssbzn) [inline mongodbatlas/cloud_provider_snapshot_export_buckets.go:162]: [nit]
```suggestion
	IAMRoleID     string `json:"iamRoleId,omitempty"`     // Unique identifier of the role that Atlas can use to access the bucket. If necessary, use the UI or API to retrieve the role ID. You must also specify the bucketName.
```

File: mongodbatlas/cloud_provider_snapshot_export_buckets.go [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+type CloudProviderSnapshotExportBuckets struct {
+	Results    []*CloudProviderSnapshotExportBucket `json:"results,omitempty"`    // Includes one CloudProviderSnapshotExportBucket object for each item detailed in the results array section.
+	Links      []*Link                              `json:"links,omitempty"`      // One or more links to sub-resources and/or related resources.
+	TotalCount int                                  `json:"totalCount,omitempty"` // Count of the total number of items in the result set. It may be greater than the number of objects in the results array if the entire result set is paginated.
+}
+
+// CloudProviderSnapshotExportBucket represents one cloud provider snapshot export buckets.
+type CloudProviderSnapshotExportBucket struct {
+	ID            string `json:"_id,omitempty"`           // Unique identifier of the S3 bucket.
+	ProjectID     string `json:"projectId,omitempty"`     // Unique 24-hexadecimal digit string identifying the project.
```
Comment (gssbzn) [inline mongodbatlas/cloud_provider_snapshot_export_buckets.go:159]: [q] this is only documented for the create, but even so the example doesn't use it, is this really necessary?

Comment (abner-dou) [reply mongodbatlas/cloud_provider_snapshot_export_buckets.go:159]: I can remove it, but I don't know if the end users will need it, due it's in the create documentation.

Comment (gssbzn) [reply mongodbatlas/cloud_provider_snapshot_export_buckets.go]: if you have tested the `curl`s and it does actually show or is needed we keep it, I'm just wondering if this is a documentation error, not a blocker to keep

**Review by gssbzn — APPROVED** [review]
LGTM

**Review by thetonymaster — APPROVED** [review]
LGTM after Gus comments

**Comment by abner-dou** [comment]
@gssbzn @thetonymaster can you merge this please

**Comment by gssbzn** [comment]
Done

**Pull request merged**